### ISSUE TITLE
only download but not replace automatically

### ISSUE THREAD

**Issue by gh279760559** [Sergei-Korneev/obsidian-local-images-plus#18]
Hi there,

thanks 1stly for this powerful plugin. Just curious will this plugin replace the online link in the MD automatically or I need to replace manually which would be a huge work. 

Cheers,
L

**Comment by gh279760559** [comment]
update:
I use ctrl + P to call the command and then the "Local image plus: Download all media files". The image is downloaded into the right folder. But in the article it does not replace the online link.

**Comment by Sergei-Korneev** [comment]
Hi. If it is a bug report you gotta be more specific (screenshots / URLs / etc).  

See how to write bug report
https://marker.io/blog/how-to-write-bug-report

**Comment by gh279760559** [comment]
> Hi. If it is a bug report you gotta be more specific (screenshots / URLs / etc).  
> 
> See how to write bug report
> https://marker.io/blog/how-to-write-bug-report

Hi, dunno if it is a bug or not. Just asking if the plugin supports this function.

**Comment by Sergei-Korneev** [comment]
> Hi, dunno if it is a bug or not. Just asking if the plugin supports this function.

Sure thing. Just an example:

![image](https://user-images.githubusercontent.com/60899505/232201685-4fec83fd-aff2-4c28-96d9-0f0c075c6625.png)

![image](https://user-images.githubusercontent.com/60899505/232201722-e57a88dc-9424-4d61-9216-8df7728224d1.png)

**Comment by gh279760559** [comment]
hmm interesting that I can see the pic location is converted automatically. Well, what I have tried are:
1. open a MD file contains some pictures (links are from website)
2. ctrl + p and select download medias...
3. images are downloaded automatically and I can see downloaded images locally and names are like the MD5 shown.
4. Unfortunately pictures in the MD file remain unchanged.
Is there anything wrong I did? Cheers.
--
Sent from myMail for Android Saturday, 15 April 2023, 04:43pm +08:00 from Sergei-Korneev  ***@***.*** :

**Comment by gh279760559** [comment]
> hmm interesting that I can see the pic location is converted automatically.

I mean I can see the conversion works in your reply. But not for me the case.

**Comment by gh279760559** [comment]
![Obsidian_5tH9K9CxuB](https://user-images.githubusercontent.com/20475927/232404349-68ff97a1-e24f-48fe-a9da-ed283ecfea20.gif)
a gif will help explain what I meant:
I tried to copy and paste a md file into the obsidian vault. You can see the plugin works and it automatically downloaded all images used in the md file. 
*I thought the corresponding images path in the md file will be converted to local path automatically. But the it does work.*

**Comment by Sergei-Korneev** [comment]
Well, now you bug report  is a little bit better, for I can see the screencast. But It will be more helpfull if you send me those files so I could test them in my local obsidian lab.

**Comment by Sergei-Korneev** [comment]
BTW, is it a cloud storage on the screencast? the plugin has not been tested for this.

**Comment by Sergei-Korneev** [comment]
![image](https://user-images.githubusercontent.com/60899505/232463316-e65f812a-4d45-4581-947d-fee4e8b39a59.png)

**Comment by gh279760559** [comment]
Yeah, it is a network storage path. Hmm, I think you are right. Actually not only this file but all others which saved there cannot be converted. Thanks man. 

If it is the case, I think the best way to solve is to contact the obsidian team to see if there is any possible solution right.

This is the file in the gif shown.
[3.二元语法与中文分词.md](https://github.com/Sergei-Korneev/obsidian-local-images-plus/files/11258759/3.md)

**Comment by gh279760559** [comment]
Update: ok, I think the NAS path is not the issue. I did a quick test:
![Obsidian_2eGHwLMQbp](https://user-images.githubusercontent.com/20475927/232696754-acb6a304-c5bb-41a2-8a79-223ebed960f6.gif)

use a local path instead of the NAS path and choose a different MD file but still no luck.

**Comment by Sergei-Korneev** [comment]
Just tested it. Works fine on my local pc.
![image](https://user-images.githubusercontent.com/60899505/232730294-70191e9c-fe98-4072-b6b1-2171aacd536e.png)

**Comment by gh279760559** [comment]
> 

Alright, finally I sort it out.
It does not work with this plugin. Thank you Pal.
![image](https://user-images.githubusercontent.com/20475927/233782945-5a7c8781-ca21-4a32-a464-2ac7bf900d44.png)

**Comment by Sergei-Korneev** [comment]
Affirmative. Thank you for your testing. Will think how to rule this out.
